enumerable.rb

```diff
@@ -0,0 +1,257 @@
+# rubocop:disable Style/CaseEquality, Style/For, Lint/RedundantCopDisableDirective, Lint/MissingCopEnableDirective, Metrics/ModuleLength, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    arr = self
+    if block_given?
+      for i in arr do
+        yield i
+      end
+    else
+      arr.to_enum
+    end
+  end
+
+  def my_each_with_index
+    arr = self
+    if block_given?
+      j = 0
+      for i in arr do
+        yield i, j
+        j += 1
+      end
+    else
+      arr.to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      arr = self
+      arr.my_each do |i|
+        new_arr.push(i) if yield i
+      end
+      new_arr
+    else
+      arr.to_enum
```

Comment:
* [ ] Pass an argument to to_enum, ie to_enum(:name_of_your_method), in this case to_enum(:my_select)
after that compare answers.
```p my_array.my_select```  and ```p my_array.select```